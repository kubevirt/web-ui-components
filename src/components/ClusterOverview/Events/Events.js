import React from 'react';

import PropTypes from 'prop-types';

import {
  DashboardCard,
  DashboardCardBody,
  DashboardCardHeader,
  DashboardCardTitle,
  DashboardCardTitleHelp,
} from '../../Dashboard/DashboardCard';
import EventsBody from '../../Dashboard/Events/EventsBody';
import { ClusterOverviewContextGenericConsumer } from '../ClusterOverviewContext';

export class Events extends React.PureComponent {
  render() {
    const { Component } = this.props;
    return (
      <DashboardCard>
        <DashboardCardHeader className="kubevirt-events__card-header">
          <DashboardCardTitle>Cluster Events</DashboardCardTitle>
          <DashboardCardTitleHelp>help for events</DashboardCardTitleHelp>
        </DashboardCardHeader>
        <DashboardCardBody id="events-body" className="kubevirt-events__card-body">
          <EventsBody>
            <Component />
          </EventsBody>
        </DashboardCardBody>
      </DashboardCard>
    );
  }
}
Events.defaultProps = {
  Component: React.Fragment,
};

Events.propTypes = {
  Component: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};

export const EventsConnected = () => <ClusterOverviewContextGenericConsumer Component={Events} dataPath="eventsData" />;

EventsConnected.propTypes = {
  ...Events.propTypes,
};

EventsConnected.defaultProps = {
  ...Events.defaultProps,
};
