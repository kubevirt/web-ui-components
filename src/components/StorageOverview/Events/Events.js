import React from 'react';

import PropTypes from 'prop-types';

import {
  DashboardCard,
  DashboardCardBody,
  DashboardCardHeader,
  DashboardCardTitle,
  DashboardCardTitleHelp,
} from '../../Dashboard/DashboardCard';
import EventsBody from '../../Dashboard/Events/EventsBody';
import { StorageOverviewContextGenericConsumer } from '../StorageOverviewContext';

export const Events = ({ Component }) => (
  <DashboardCard>
    <DashboardCardHeader className="kubevirt-events__card-header">
      <DashboardCardTitle>Events</DashboardCardTitle>
      <DashboardCardTitleHelp>help for events</DashboardCardTitleHelp>
    </DashboardCardHeader>
    <DashboardCardBody id="events-body" className="kubevirt-events__card-body">
      <EventsBody>
        <Component />
      </EventsBody>
    </DashboardCardBody>
  </DashboardCard>
);
Events.defaultProps = {
  Component: React.Fragment,
};

Events.propTypes = {
  Component: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};

const EventsConnected = () => <StorageOverviewContextGenericConsumer Component={Events} dataPath="eventsData" />;

export default EventsConnected;
