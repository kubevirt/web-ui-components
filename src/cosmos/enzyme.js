import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
