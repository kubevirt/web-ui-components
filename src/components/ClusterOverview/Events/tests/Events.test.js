import React from 'react';
import { render } from 'enzyme';

import { Events } from '../Events';
import { default as EventsFixtures } from '../fixtures/Events.fixture';

const testEventsOverview = () => <Events {...EventsFixtures.props} />;

describe('<Events />', () => {
  it('renders correctly', () => {
    const component = render(testEventsOverview());
    expect(component).toMatchSnapshot();
  });
});
