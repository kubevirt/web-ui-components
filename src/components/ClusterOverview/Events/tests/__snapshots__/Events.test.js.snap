// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Events /> renders correctly 1`] = `
<div
  class="card-pf kubevirt-dashboard__card"
>
  <div
    class="card-pf-heading kubevirt-dashboard__card-heading"
  >
    <h2
      class="card-pf-title"
    >
      Cluster Events
    </h2>
    <button
      class="btn btn-link"
      type="button"
    >
      <span
        aria-hidden="true"
        class="fa fa-info-circle fa-2x"
      />
    </button>
  </div>
  <div
    class="card-pf-body kubevirt-events__body"
    id="events-body"
  >
    <div>
      Requires component from tectonic
    </div>
  </div>
</div>
`;
